### DISCUSSION TITLE
viem `WalletClient` type incompatiability with `hardhat-viem` functions

### DISCUSSION THREAD

**Discussion by 0xCourtney** [NomicFoundation/hardhat#5118] [Troubleshooting] [answered]
I'm struggling to get the following code to work I want to be able to pass the WalletClient as a function parameter however I'm getting type incompatibility errors.

```typescript
export async function deploy(
  deployer: WalletClient,
  contractName: string,
  constructorArgs: any[],
) {
  const { deploymentTransaction, contract } =
    await viem.sendDeploymentTransaction(contractName, constructorArgs, {
      client: {
        wallet: deployer, // Types of property account are incompatible. Type Account | undefined is not assignable to type Account. Type undefined is not assignable to type Account.
      },
    });

   [...]
}
```

Calling `getWalletClient` within the function and using the returned value in the config works fine, however, I don't want to do this for every single helper function.

**Comment by schaable** [comment]
Are you importing `WalletClient` from `viem` or `hardhat-viem`? Can you share a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Reply by 0xCourtney** [reply]
`WalletClient` is imported from `viem` if I try to import from `hardhat-viem` I get `Module "@nomicfoundation/hardhat-viem" has no exported member WalletClient`. The code below should be enough to reproduce the issue.

``` typescript
// ./scripts/deploy.ts

import { WalletClient } from 'viem';

export async function deploy(
  deployer: WalletClient,
  contractName: string,
  constructorArgs: any[],
) {
  const { deploymentTransaction, contract } =
    await viem.sendDeploymentTransaction(contractName, constructorArgs, {
      client: {
        wallet: deployer, // Types of property account are incompatible. Type Account | undefined is not assignable to type Account. Type undefined is not assignable to type Account.
      },
    });
}
```

**Reply by schaable** [reply, marked as answer]
Try with `import { WalletClient } from "@nomicfoundation/hardhat-viem/types";`

**Reply by 0xCourtney** [reply]
Thanks! Idk how I missed that.
